src/lib.rs

```diff
@@ -88,6 +89,40 @@ where
     }
 }
 
+/// Result of a value-to-value conversion with post-validation of the output value
+pub type ValidatedResult<V> = CoreResult<V, (V, Context<<V as Validate>::Invalidity>)>;
+
+/// Value-to-value conversion with post-validation of the output value
+///
+/// The validation is performed on the output value after the input value
+/// has been consumed during the conversion.
+///
+/// On success the output value is returned. On validation errors the output
+/// value is returned together with all invalidities.
+pub trait ValidatedFrom<T>: Validate + Sized {
+    /// Perform the validation
+    fn validated_from(from: T) -> ValidatedResult<Self>;
```

Comment:
Probably `validate_from` would suite better here?

Reply:
I wanted to keep the naming of both traits and their functions as consistent as possible:
- the term _validated_ emphasizes that this is a post-validation, i.e. performed on the output value
- `validated_from()` closely matches `into_validated()`
- both `validated_from()` as well as `validate_from()` produce readable code
- renaming `into_validated()` as `into_validate()` does not work for me

One option would be to rename both traits and functions:
- `ValidateInto` / `validate_into()`
- `ValidateFrom` / `validate_from()`

Reply:
What about `from_validated` and `into_validated`?